### ISSUE TITLE
Hang in MacOS Sequoia after running AppleScript

### ISSUE THREAD

**Issue by hrydgard** [create-dmg/create-dmg#185]
After "Running AppleScript to make Finder stuff pretty", it now just hangs. See log below.

(Please ignore the array syntax of the printed command line, it's just how my wrapper app prints the list of arguments before running it quietly).

```
create-dmg ["--volname", "PPSSPP", "--window-pos", "200", "120", "--window-size", "800", "400", "--icon-size", "100", "--icon", "PPSSPPSDL.app", "200", "190", "--hide-extension", "PPSSPPSDL.app", "--app-drop-link", "600", "185", "PPSSPP_macOS.dmg", "/Users/hrydg/temp/bot-temp/dmgtemp/"]

Creating disk image...
...............................................................................................................................................................................
created: /Users/hrydg/temp/bot-temp/rw.12397.PPSSPP_macOS.dmg
Mounting disk image...
Device name:     /dev/disk12
Searching for mounted interstitial disk image using /dev/disk12s... 
Mount dir:       /Volumes/dmg.brQozN
Making link to Applications dir...
/Volumes/dmg.brQozN
Running AppleScript to make Finder stuff pretty: /usr/bin/osascript "/var/folders/wj/zk92qzvj4910x07p06whmcnh0000gn/T/createdmg.tmp.XXXXXXXXXX.ZikiyjabHt" "dmg.brQozN"
```

**Comment by iSapozhnik** [comment]
Same here. The good thing that it does the job in the end but takes a lot of time:
```
Running AppleScript to make Finder stuff pretty: /usr/bin/osascript "/var/folders/c7/vh7wn5k135x5jh9jvf77f1pc0000gn/T/createdmg.tmp.XXXXXXXXXX.bRj6eyb8Zz" "dmg.1qQRya"
waited 412 seconds for .DS_STORE to be created.
Done running the AppleScript...
```

**Comment by pzhlkj6612** [comment]
Were there log messages in Console during the execution of AppleScript? Did you get high CPU or memory usage?

Have no mac for test now, though.

**Comment by hrydgard** [comment]
Mine also finished on the next try. No logging message, just a randomly long delay, it seems.

**Comment by git-n-pissed** [comment]
I don't think it is random.  I've found if you open a new Finder window once the script appears to have hung, it will un-hang immediately.

The problem is that [this line](https://github.com/create-dmg/create-dmg/blob/32121505917f6eb83c9a5144a80e4f5c95557fef/support/template.applescript#L70) will not set the `ejectMe` condition to `true` until Finder creates the `.DS_STORE` file, but it isn't creating it.

I don't know for sure what the problem is with [support/template.applescript](https://github.com/create-dmg/create-dmg/blob/32121505917f6eb83c9a5144a80e4f5c95557fef/support/template.applescript) which is causing Finder to not complete the intended task, but I suspect a race condition has been introduced with use of calls to `delay` between critical sections of the script.

**Comment by iSapozhnik** [comment]
Ah yes forgot to update the thread. I can confirm that opening a new Finder window does the job.

**Comment by zaneclaes** [comment]
Unfortunately, opening a new finder window does not fix the problem for me (MacOS 15.5, create-dmg v1.2.1). It works about half of the time, but the other half nothing I do seems to un-stick the script. I've also tried closing the window and deleting the tmp dmgs. Looks like I'm going to need to use `--sandbox-safe` to bypass the apple script entirely :(
